utils/warnings.ml

```diff
@@ -814,6 +818,12 @@ let message = function
     "This pattern depends on mutable state.\n\
      It prevents the remaining arguments from being uncurried, which will \
      cause additional closure allocations."
+  | Use_generic_comparison msg ->
+      let msg = match msg with
+        | Some msg -> "\nNote: "^msg
+        | None -> ""
+      in
+      "Use of generic comparison functions are discouraged."^msg
```

Comment:
If somebody has a better idea on what to tell the user to convince them, I would gladly take it.